### DISCUSSION TITLE
Trouble receiving data via HTTP on OpenTelemetry Collector in Docker

### DISCUSSION THREAD

**Discussion by ZziM** [open-telemetry/opentelemetry-collector#9316] [Q&A]
I am working on integrating OpenTelemetry .NET SDK with Elastic APM and am using OpenTelemetry Collector to forward traces and metrics.

My service is running in Docker, and I'm using otel/opentelemetry-collector-contrib:0.91.0 as the collector. I've set up the collector to receive data over both gRPC and HTTP as indicated in the screenshots provided.

Why are the data sent via HTTP to port 4318 not appearing in the collector's logs, while everything works fine over gRPC on port 4317? I've verified that the ports are open and accessible. What could be the issue?

When I configure the .NET SDK to send data over gRPC to port `4317`, everything works as expected, and I see the data in the collector's logs. However, when I switch the configuration to send data over HTTP to port `4318`, I do not see any data in the logs, even though the logs clearly show that the HTTP server has started and is listening on port `4318`.

More detailed info you can see on my [stackoverflow question](https://stackoverflow.com/questions/77839599/trouble-receiving-data-via-http-on-opentelemetry-collector-in-docker)

Here is the relevant portion of my otel-collector.yml for the collector:
receivers:
  otlp:
    protocols:
      grpc:
        endpoint: localhost:4317
      http:
        endpoint: localhost:4318

Maybe somebody can help with this issue?

**Comment by srinivasmummareddy** [comment]
@ZziM  Even I'm facing similar issue...Did you find the root cause?
